pybossa/api/__init__.py

```diff
@@ -713,3 +713,37 @@ def _get_valid_service(task_id, service_name, payload, proxy_service_config):
         'Task {} loaded for user {} failed calling {} service with payload {}'.format(task_id, current_user.id, service_name, payload))
 
     return abort(403, 'The request data failed validation')
+
+
+@jsonpify
+@login_required
+@csrf.exempt
+@blueprint.route('/task/<int:task_id>/assign', methods=['POST'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def assign_task(task_id=None):
+    """Assign task to users for locked, user_pref and task_queue schedulers."""
+    projectname = request.json.get('projectname', None)
+    a_project = project_repo.get_by_shortname(projectname)
+    if not a_project:
+        return abort(400)
```

Comment:
This line can also report back that 400 with suitable message something similar to following

```python
 if not a_project:
        return abort(400, f"Invalid project name {projectname}")
```